server/server.py

```diff
@@ -0,0 +1,566 @@
+            print("Had an error while encrypting AES key with RSA public key. Possibly bad RSA key.")
+            failed_recon_pckt = ResourceWarning.ReconnectionDeclinedPacker(SERVER_VER, clients_dict[client_name].id)
+            conn.sendall(failed_recon_pckt)
+            return ERROR
+        return GENERAL_ERROR
+
+    # Select the appropriate packet type based on registration or reconnection
+    if is_reg:
+        pubkey_pckt = ResponsePacker.EncryptedAESKeyRegistrationPacker(SERVER_VER, clients_dict[client_name].id, enc_aes_key)
+    else:
+        pubkey_pckt = ResponsePacker.EncryptedAESKeyReconnectionPacker(SERVER_VER, clients_dict[client_name].id, enc_aes_key)
+    send_packet(conn, pubkey_pckt.pack())
+
+    # Update the database with the newly generated AES key
+    with sqlite3.connect("defensive.db") as conn_db:
+        cursor = conn_db.cursor()
+        cursor.execute("UPDATE clients SET AES_key = ? WHERE ID = ?", (aes_key, clients_dict[client_name].id))
+        conn_db.commit()
+
+    print(f"Public key accepted, encrypted AES key sent.")
+    return SUCCESS
+
+# **********************************************************************
+# File recv related functions
+# **********************************************************************
+def get_enc_file_content(conn, file_name, total_packets, enc_file_size, file_size, enc_file_content):
+    """
+    Receives the encrypted file content by collecting packets from the connection.
+    Validates each packet's code, packet number, and file name for integrity.
+    
+    Args:
+        conn: The connection object.
+        file_name (str): The expected name of the file being received.
+        total_packets (int): Total number of packets expected.
+        enc_file_size (int): Expected size of the encrypted file.
+        file_size (int): Expected size of the decrypted file.
+        enc_file_content (str): Placeholder for accumulating file content.
+        
+    Returns:
+        str: The full encrypted file content if successful; otherwise, 0 on error.
+    """
+    for curr_packet in range(2, total_packets + 1): 
+        # Receive the next packet and validate essential fields
+        packet = recv_packet(conn)
+        if packet['code'] != 828:
+            print(f"Error: Invalid packet code: ({packet['code']} instead of 828)")
+            return 0
+        if packet['packet_number'] != curr_packet:
+            print(f"Error: Invalid packet number ({packet['packet_number']} instead of {curr_packet})")
+            return 0
+        
+        # Check if the file name matches the expected name
+        packet_file_name = packet['file_name'].split('\\')[-1]
+        if packet_file_name != file_name:
+            print(f"Error: invalid file name ({packet_file_name} instead of {file_name})")
+            return 0
+
+        # Append current packet's content to the complete encrypted file content
+        enc_file_content += packet['message_content']

```

Comment:
aren't you supposed to update the db every iteration? keep the file in the db but unverified until completed?